lib/common.sh

```diff
@@ -638,31 +638,29 @@ common::argc_validate () {
   fi
 }
 
-
 ###############################################################################
-# Get the md5 hash of a file
+# Get the SHA hash of a file
 # @param file - The file
-# @print the md5 hash
-common::md5 () {
-  local file=$1
-
-  if which md5 >/dev/null 2>&1; then
-    md5 -q "$1"
-  else
-    md5sum "$file" | awk '{print $1}'
-  fi
-}
-
-###############################################################################
-# Get the sha1 hash of a file
-# @param file - The file
-# @param algo - Algorithm 1 (default), 224, 256, 384, 512, 512224, 512256
+# @param algo - Algorithm 1, 224, 256 (default), 384, 512, 512224, 512256
+# output_type - Specifies output for the SHA:
+#               hash (default): outputs just the SHA
+#               file: outputs the SHA, two spaces, and the basename of the file
 # @print the sha hash
 common::sha () {
   local file=$1
-  local algo=${2:-1}
+  local algo=${2:-256}
+  local output_type=${3:-hash}
+  local shasum_output
+
+  which shasum >/dev/null 2>&1 || return 1
```

Comment:
I think there are several places in k8s where we ignore this shellcheck, because `command -v` may not be the right approach.
@kubernetes/bash-firefighters -- can you advise?

Reply:
short version: `command -v` is not equivilant, when you want a binary use `which`

Reply:
long version: `command -v` will consider functions in the shell, the reason this could be bad to use over `which` is if you are checking that a binary is installed for some other layer to invoke. `command -v` will return true if a function matches even if there is not a matching binary. `which` only inspects files in `PATH`.

https://github.com/kubernetes/kubernetes/blob/master/hack/verify-shellcheck.sh#L39
https://github.com/koalaman/shellcheck/issues/1162